### PR TITLE
[WASM] Support LinearGradientBrush on Shapes

### PR DESCRIPTION
**Pull request by davidjohnoliver**
Maps LinearGradientBrush to a linearGradient element nested inside the Shape's SVG.

Note that angled gradients don't match the angle on UWP (a similar issue is present on iOS).

GitHub Issue (If applicable): #

## PR Type

- Feature

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Media/LinearGradientBrush.wasm.cs [hunk 1]
```diff
@@ -23,5 +24,26 @@ internal string ToCssString(Size size)
...
+		/// </summary>
+		internal UIElement ToSvgElement()
+		{
+			var linearGradient = new SvgElement("linearGradient");
+
+			linearGradient.SetAttribute(
+				("x1", StartPoint.X.ToString()),
+				("y1", StartPoint.Y.ToString()),
+				("x2", EndPoint.X.ToString()),
+				("y2", EndPoint.Y.ToString())
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Media/LinearGradientBrush.wasm.cs]: They need the `Invariant` number format.

File: src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs [hunk 2]
```diff
@@ -67,11 +71,23 @@ protected virtual void OnChildrenChanged()
...
+					_fillBrushSubscription.Disposable = null;
+					break;
+				case LinearGradientBrush lgb:
+					var linearGradient = lgb.ToSvgElement();
+					var gradientId = linearGradient.HtmlId;
+					SvgChildren.Add(linearGradient);
+					svgElement.SetStyle("fill", $"url(#{gradientId})");
+					_fillBrushSubscription.Disposable = new DisposableAction(
+						() => SvgChildren.Remove(linearGradient)
+					);
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs]: You should also support "Stroke", not just "Fill".

File: src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs [hunk 3]
```diff
@@ -144,6 +144,12 @@ private Rect GetBBoxOfChildrenWithStrokeThickness()
 
 			foreach (FrameworkElement child in GetChildren())
 			{
+				if (child.HtmlTag == "linearGradient")
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs]: You should put this in a `<defs>` tag an just filter-out this one.

**Comment by carldebilly** [comment]
A sample is missing.  You can take the sample from my branch :-)
https://github.com/unoplatform/uno/commits/dev/cdb/svg-gradient

**Comment by davidjohnoliver** [comment]
> A sample is missing. You can take the sample from my branch :-)

I added the sample when I did iOS yesterday :)

File: src/Uno.UI/UI/Xaml/Media/LinearGradientBrush.wasm.cs (hunk 1 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Media/LinearGradientBrush.wasm.cs]: fixed

File: src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs (hunk 2 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.wasm.cs]: Done!

I noticed that the Stroke widths were off, but this seems to be a known issue (https://github.com/unoplatform/uno/issues/1552).

File: src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs (hunk 3 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Shapes/ArbitraryShapeBase.wasm.cs]: done

**Pull request merged**